src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -246,6 +246,8 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
         REMOVE_BLOBS,
         /// is set when Clickhouse is sure that the blobs belong to other replica and current replica has not locked them on s3 yet
         PRESERVE_BLOBS,
+        /// remove blobs even if the part is not temporary
+        REMOVE_BLOBS_OF_NOT_TEMPORARY,
```

Comment:
What do you think if we rename all enum cases?

REMOVE_BLOBS -> REMOVE_BLOBS_WHEN_TEMPORARY_STATE
PRESERVE_BLOBS -> PRESERVE_BLOBS_WHEN_TEMPORARY_STATE
REMOVE_BLOBS_OF_NOT_TEMPORARY -> REMOVE_BLOBS_WHEN_ANY_STATE

But I do not insist at all. May be it is ok but in separate PR, because that is the fix.

Reply:
I can do it in a separate PR, but I don't really care. The enum name (`BlobsRemovalPolicyForTemporaryParts`) implies that it's for temporary parts. Maybe I should not have added the new value to this enum (because the new value is used for non-temporary parts), but I didn't want to introduce another special bool flag in `IMergeTreeDataPart`